packages/git/src/browser/diff/git-diff-service.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { GitLogOptions } from './git-diff-model';
+import { injectable, inject } from "inversify";
+import { GitRepositoryProvider } from '../git-repository-provider';
+import { Git, Repository } from '../../common';
+import { FileUri } from '@theia/core/lib/node/file-uri';
+import * as Path from 'path';
+
+@injectable()
+export class GitDiffService {
+
+    constructor(
+        @inject(Git) protected readonly git: Git,
+        @inject(GitRepositoryProvider) protected readonly repositoryProvider: GitRepositoryProvider
+    ) { }
+
+    protected toUri(repository: Repository, pathSegment: undefined): undefined;
+    protected toUri(repository: Repository, pathSegment: string): string;
+    protected toUri(repository: Repository, pathSegment: string | undefined): string | undefined {
+        if (pathSegment === undefined) {
+            return undefined;
+        }
+        return FileUri.create(Path.join(FileUri.fsPath(repository.localUri), pathSegment)).toString();
+    }
+
+    protected getRangeArg(options?: GitLogOptions): string {
```

Comment:
Unused module? At least ought to be.

Reply:
Yes, I am positive. It is obsolete. A service class without a single public method is most likely unused.

Reply:
You're right. I shouldn't be too attached to such useless things.